src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -431,11 +432,9 @@ def __init__(  # pylint: disable=too-many-arguments
         self._external_id_match = external_id_match
         self._verify_timeout = verify_timeout
 
-    def snapshot(self) -> list[MigratedGroup]:
-        return self._snapshot(self._fetcher, self._crawler)
-
     def has_groups(self) -> bool:
-        return len(self.snapshot()) > 0
+        first_element = islice(self.snapshot(), 1)
```

Comment:
can you revert the change on this method? i find it risky. it's fine to load snapshot twice, as it's materialized

Reply:
I've reverted it.

This was intended to be a poor-man's `limit 1` because `.snapshot()` is normally lazy, etc.

Reply:
Pfft. And I just noticed the method isn't actually called from anywhere (including tests). So I'm just going to nuke it entirely.

Reply:
… and removed.